vision/objectdetection/color_detector.go

```diff
@@ -11,7 +11,9 @@ import (
 // ColorDetectorConfig specifies the fields necessary for creating a color detector.
 type ColorDetectorConfig struct {
 	SegmentSize       int     `json:"segment_size_px"`
-	Tolerance         float64 `json:"tolerance_pct"`
+	HueTolerance      float64 `json:"hue_tolerance_pct"`
+	SaturationCutoff  float64 `json:"saturation_cutoff_pct"`
+	ValueCutoff       float64 `json:"value_cutoff_pct"`
```

Comment:
Added two new parameters here, and renamed Tolerance to be clearer about what it actually represents.